symengine/sets.cpp

```diff
@@ -935,7 +925,27 @@ RCP<const Set> conditionset(const RCP<const Basic> &sym,
                     if (not(is_a_Number(*elem) or is_a<Constant>(*elem))) {
                         others.insert(elem);
                     } else {
-                        present.insert(elem);
+                        // logical_and() doesn't guarantee that if element of a
```

Comment:
@isuruf , can you have a look here?
I added a test you suggested for this below.